src/app/[...mdxPath]/page.tsx

```
@@ -0,0 +1,27 @@
+
+interface Props {
+  params: { mdxPath: string[] };
+}
+
+export async function generateMetadata(props: Props) {
+  const params = await props.params;
+  const { metadata } = await importPage(params.mdxPath);
+  return metadata;
+}
+
+const Wrapper = useMDXComponents({}).wrapper;
```

ESLint

[error] React Hook "useMDXComponents" cannot be called at the top level. React Hooks must be called in a React function component or a custom React Hook function.

(react-hooks/rules-of-hooks)